src/normalize/glossary.ts

```diff
@@ -0,0 +1,202 @@
+/// Copyright 2020 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+import { Editor, Node, NodeEntry, Path, Transforms } from 'slate'
+
+import { Definition, DefinitionTerm, DefinitionExample, Glossary, Meaning, SeeAlso, Term } from '../interfaces'
+import { normalizeOrderedChildren, previousOverOnly } from './util'
+
+/**
+ * Normalize glossary
+ *
+ * Return true if entry was normalized and false otherwise.
+ */
+export default function normalizeGlossary(editor: Editor, entry: NodeEntry): boolean {
+    const [node, path] = entry
+
+    if (Glossary.isGlossary(node)) {
+        // Glossary must not be empty.
+        if (Editor.isEmpty(editor, node)) {
+            Transforms.removeNodes(editor, { at: path })
+            return true
+        }
+
+        // Glossary must be the last element.
+        const [next, nextPath] = Editor.next(editor, { at: path }) || []
```

Comment:
[BLOCKER]
Currently, it is possible for `glossary` to be inside a `section`. It should not happen because then it will not be properly serialized.
I'm not sure what is the best place to handle this case. Here or in `structure.ts`